tests/handlers/test_profile.py

```diff
@@ -147,3 +161,20 @@ def test_set_my_avatar(self):
             (yield self.store.get_profile_avatar_url(self.frank.localpart)),
             "http://my.server/pic.gif",
         )
+
+    @defer.inlineCallbacks
+    def test_set_my_avatar_if_disabled(self):
+        self.hs.config.disable_set_avatar_url = True
+
+        # Set first time avatar is allowed, if displayname is null
```

Comment:
You probably meant that setting an avatar for the first time is allowed if this user doesn't have an avatar, not if this user doesn't have a display name.